src/kiwixapp.cpp

```diff
@@ -208,11 +208,21 @@ void KiwixApp::restoreTabs()
       {
         try
         {
+          if (zimUrl == "SettingsTab")
+          {
+            getTabWidget()->openOrSwitchToSettingsTab();
+            continue;
+          }
+
           /* Throws exception if zim file cannot be found */
           m_library.getArchive(QUrl(zimUrl).host().split('.')[0]);
           openUrl(QUrl(zimUrl));
         }
-        catch (std::exception &e) { /* Blank */ }
+        catch (std::exception &e)
+        { 
+          /* Open an empty tab for tabs that could not be loaded. */
+          openUrl(QUrl(""));
+        }
```

Comment:
This will result in the URL of the tab being lost. This is undesirable if the ZIM files are located on removable storage that was not mounted at the time when `kiwix-desktop` was launched.

A better option may be to open the tab with an error text in it.

@kelson42 What do you think?

Reply:
Yes it should open the right link and print a proper error message (depending which exception is triggered)

Reply:
Done.